plugins/eos/gs-plugin-eos.c

```diff
@@ -2068,31 +2107,59 @@ static void
 process_proxy_updates (GsPlugin *plugin,
 		       GsAppList *list,
 		       GsApp *proxy_app,
+		       gboolean install_missing,
 		       const char **proxied_apps)
 {
-	g_autoptr(GSList) proxied_updates = NULL;
+	g_autoptr(GsAppList) proxied_updates = gs_app_list_new ();
+
+	GsApp *cached_proxy_app = gs_app_list_lookup (list,
+						      gs_app_get_unique_id(proxy_app));
+	if (cached_proxy_app != NULL)
+		gs_app_list_remove (list, cached_proxy_app);
+
+	/* add all the apps if we should install the missing ones; the real sorting for
+	 * whether they're already installed is done in the refine method */
+	if (install_missing) {
+		for (guint i = 0; proxied_apps[i] != NULL; ++i) {
+		        g_autoptr(GsApp) app = gs_app_new (proxied_apps[i]);
+			gs_app_add_quirk (app, GS_APP_QUIRK_IS_WILDCARD);
+			gs_app_set_bundle_kind(app, AS_BUNDLE_KIND_FLATPAK);
+
+			gs_app_list_add (proxied_updates, app);
+		}
+	}
 
 	for (guint i = 0; i < gs_app_list_length (list); ++i) {
 		GsApp *app = gs_app_list_index (list, i);
+		GsApp *added_app = NULL;
 		const char *id = gs_app_get_id (app);
 
 		if (!g_strv_contains (proxied_apps, id) ||
 		    gs_app_get_scope (proxy_app) != gs_app_get_scope (app))
 			continue;
 
-		proxied_updates = g_slist_prepend (proxied_updates, app);
+		added_app = gs_app_list_lookup (proxied_updates, gs_app_get_unique_id(app));
+		if (added_app == app)
+			continue;
+
+		gs_app_list_remove (proxied_updates, added_app);
+
+		gs_app_set_state (app, AS_APP_STATE_UPDATABLE_LIVE);
+		gs_app_list_add (proxied_updates, app);
 	}
 
 	if (!proxied_updates)
```

Comment:
This will now never be true; you may want to check the length of the list instead?

Reply:
Great catch!